framework.py

```diff
@@ -17,8 +18,45 @@ class ShotgunUtilsFramework(sgtk.platform.Framework):
             
     def init_framework(self):
         self.log_debug("%s: Initializing..." % self)
+        self._bg_cleanup_thread = None
+        self._post_old_data_cleanup()
     
     def destroy_framework(self):
         self.log_debug("%s: Destroying..." % self)
+        if self._bg_cleanup_thread:
+            if self._bg_cleanup_thread.isAlive():
+                # If the clean up is not completed yet, log why we are waiting.
+                self.log_info("Waiting for old data clean up to complete...")
+            self._bg_cleanup_thread.join()
     
-    
+    def _post_old_data_cleanup(self):
+        """
+        If the current tk-core version supports it, and running in tk-desktop,
+        post a cleanup of old data in the background.
+        """
+        # This Bundle method was introduced in tk-core > v0.18.124
+        # Check the existence of the attribute to not introduce a dependency
+        # on tk-core release.
+        if not hasattr(self, "_remove_old_cached_data"):
+            return
+        # Cleaning up data can be slow if a lot of old data is present. As a
+        # safety measure, we only post it when running from tk-desktop, which
+        # allows us to not trigger the clean up every single time this framework
+        # is loaded.
+        if self.engine.instance_name != "tk-desktop":
+            return
+
+        try:
+            self.log_info(
+                "Posting old cached data clean up for engine %s...." % self.engine.instance_name
+            )
+            # Qt might not be yet available at this stage (e.g. in tk-desktop),
+            # so use regular Python Thread to post the clean up in the background.
+            self._bg_cleanup_thread = threading.Thread(
+                target=self._remove_old_cached_data,
```

Comment:
i am worried that we are over-thinking this. I think, at least for the time being, we should keep this out of core and inside this framework. These kind of dependencies can be a challenge to manage.